File: examples/slack/bot.js
// Connect to Slack

"use strict";

var util = require("util");
var fs = require("fs");

var slack = require("slack").slack;

function Handler() {
    this.onHello = function(hello) {
        console.log(hello);
    };
    this.onSlackError = function(error) {
        console.log(error);
    };
    this.onSlackEvent = function(event) {
        console.log(event);
    };
    this.onMessage = function(message) {
        console.log(message);
        if (message.text !== null && message.text.indexOf("quark") > -1) {
            message.channel.send("JavaScript bot says hi!");
        }
    };
}
util.inherits(Handler, slack.SlackHandler);

// you need to go to https://api.slack.com/web and generate an access token
try {
    var token = fs.readFileSync(".slack.token", "ASCII").trim();
} catch (e) {
    console.log("Failed to read Slack token. See examples/README.md for more information.");
    process.exit(1);
}
var client = new slack.SlackClient(token);
client.post("#demo", "JavaScript Slack client is connected");
client.subscribe(new Handler());

File: examples/slack/bot.py
# Connect to Slack

from quark_threaded_runtime import get_runtime
import slack


class Handler(slack.SlackHandler):

    def onHello(self, greeting):
        print "Hello received:", greeting

    def onSlackError(self, error):
        print "Error:", error.code, error.text

    def onSlackEvent(self, event):
        print "Event:", event

    def onMessage(self, message):
        print "Received:", message.text
        if message.text and "quark" in message.text:
            message.channel.send("bot.py is excited about \"%s\"" % message.text.replace("quark", "[bleep]"))

    def onMessageSent(self, message):
        print "Acknowledged:", message.text


def main():
    try:
        token = open(".slack.token").read().strip()
    except IOError:
        # you need to go to https://api.slack.com/web and generate an access token
        exit("Failed to read Slack token. See examples/README.md for more information.")

    runtime = get_runtime()
    client = slack.Client(runtime, token)
    client.post("#demo", "testing...")
    client.subscribe(Handler())
    runtime.join()


if __name__ == '__main__':
    main()
